### PR TITLE
Shared buffer fixes

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -22,5 +22,9 @@
     "target": "esnext",
     "types": []
   },
-  "files": ["node_modules/typescript/lib/lib.esnext.d.ts", "js/main.ts"]
+  "files": [
+    "node_modules/typescript/lib/lib.esnext.d.ts",
+    "node_modules/typescript/lib/lib.es2017.sharedmemory.d.ts",
```
Comment (ry) [inline tsconfig.json]: Is this line necessary? Shouldn't it be included from `lib.esnext.d.ts` ?

File: libdeno/internal.h [hunk 2]
```diff
@@ -98,7 +98,7 @@ class DenoIsolate {
   v8::Persistent<v8::Function> recv_;
   v8::StartupData snapshot_;
   v8::Persistent<v8::ArrayBuffer> global_import_buf_;
-  v8::Persistent<v8::ArrayBuffer> shared_ab_;
+  v8::Persistent<v8::SharedArrayBuffer> shared_ab_;
```
Comment (ry) [inline libdeno/internal.h]: Is there a test we could add that demonstrates it failing using ArrayBuffer? It's not clear to me why this is necessary.

Comment (piscisaureus) [reply libdeno/internal.h]: Nothing is failing, but it's not possible to use Atomics on a regular ArrayBuffer.
If you don't believe me, try this:

```
D:\deno3>node
> let ab = new ArrayBuffer(4);
undefined
> let u8 = new Uint8Array(ab)
undefined
> Atomics.add(u8, 0, 1)
TypeError: [object Uint8Array] is not an integer shared typed array.
    at Atomics.add (<anonymous>)
>
```

File: tsconfig.json (hunk 1 above)
Comment (piscisaureus) [reply tsconfig.json]: Removed this patch as it didn't seem to fully fix the problem it was supposed to fix.

File: libdeno/internal.h (hunk 2 above)
Comment (ry) [reply libdeno/internal.h]: You can modify this test, I think 
```c
TEST(LibDenoTest, Shared) {
  uint8_t s[] = {0, 1, 2};
  deno_buf shared = {nullptr, 0, s, 3};
  Deno* d = deno_new(deno_config{0, snapshot, shared, nullptr});
  EXPECT_TRUE(deno_execute(d, nullptr, "a.js", "Shared()"));
  deno_execute(d, nullptr, "a.js", "Shared()");
  EXPECT_EQ(nullptr, deno_last_exception(d));
  EXPECT_EQ(s[0], 42);
  EXPECT_EQ(s[1], 43);
  EXPECT_EQ(s[2], 44);
  deno_delete(d);
}
```

**Comment by piscisaureus** [comment]
Needs a different solution.

File: libdeno/internal.h (hunk 2 above)
Comment (piscisaureus) [reply libdeno/internal.h]: Improved (but not working):

```c++
TEST(LibDenoTest, SharedAtomics) {
  int32_t s[] = {0, 1, 2};
  deno_buf shared = {nullptr, 0, reinterpret_cast<uint8_t*>(s), sizeof s};
  Deno* d = deno_new(deno_config{0, empty, shared, nullptr});
  deno_execute(d, nullptr, "a.js",
               "Atomics.add(new Int32Array(libdeno.shared), 0, 1)");
  EXPECT_EQ(nullptr, deno_last_exception(d));
  EXPECT_EQ(s[0], 1);
  EXPECT_EQ(s[1], 1);
  EXPECT_EQ(s[2], 2);
  deno_delete(d);
}
```

**Pull request merged**